components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5541,18 +5542,31 @@ private Attribute getSchemaAttributes(ExternalClaim scimClaim, LocalClaim mapped
                                           boolean isExtensionAttr) {
 
         String name = scimClaim.getClaimURI();
+        String claimDielectURI = scimClaim.getClaimDialectURI();
+        boolean isCustomSchemaAttr = StringUtils.equalsIgnoreCase(claimDielectURI,getCustomSchemaURI());
+        boolean isComplexInAdditionalProp = false;
+        if (mappedLocalClaim != null && mappedLocalClaim.getClaimProperties() != null) {
+            for (Map.Entry<String, String> claimProperty : mappedLocalClaim.getClaimProperties().entrySet()) {
+                if (StringUtils.equalsIgnoreCase(claimProperty.getKey(), SCIMConfigConstants.DATA_TYPE)) {
```

Comment:
```suggestion
                if (SCIMConfigConstants.DATA_TYPE.equalsIgnoreCase(claimProperty.getKey())) {
```

Reply:
Isn't it a best practice using StringUtils for string comparisons?

Reply:
even under StringUtils.equalsIgnoreCase() methods also uses the same method internally. 
The risk of StringUtils.equalsIgnoreCase() is StringUtils.equalsIgnoreCase(null,null)  = true
If you are comparing a string against a constant `<Constant>.equalIgnoreCase(<variable>)` would be better.
If you are comparing two variables first variable should be checked whether it's not null to avoid NPE and use `<variable1>.equalIgnoreCase(<varibale2>)`

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/459/commits/a227f2e3af8b1ad2afaaab992d44c5f8e2b64c70